lib/ansible/modules/cloud/smartos/vmadm.py

```diff
@@ -0,0 +1,609 @@
+    filesystems:
+        required: false
+        description:
+          - Mount additional filesystems into an OS VM.
+    firewall_enabled:
+        required: false
+        description:
+          - Enables the firewall, allowing fwadm(1M) rules to be applied.
+    fs_allowed:
+        required: false
+        description:
+          - Comma separated list of filesystem types this zone is allowed to mount.
+    hostname:
+        required: false
+        description:
+          - Zone/VM hostname.
+    image_uuid:
+        required: false
+        description:
+          - Image UUID.
+    indestructible_zoneroot:
+        required: false
+        description:
+          - Adds an C(@indestructible) snapshot to zoneroot.
+    internal_metadata:
+        required: false
+        description:
+          - Metadata to be set and associated with this VM, this contains operator
+            generated keys.
+    kernel_version:
+        required: false
+        description:
+          - Kernel version to emulate for LX VMs.
+    maintain_resolvers:
+        required: false
+        description:
+          - Resolvers in C(/etc/resolv.conf) will be updated when updating
+            the C(resolvers) property.
+    max_physical_memory:
+        required: false
+        description:
+          - Maximum amount of memory (in MiBs) on the host that the VM is allowed to use.
+    max_swap:
+        required: false
+        description:
+          - Maximum amount of virtual memory (in MiBs) the VM is allowed to use.
+    nic_driver:
+        required: false
+        description:
+          - Default value for a virtual NIC model for KVM guests.
+    nics:
+        required: false
+        description:
+          - A list of nics to add, valid properties are documented in vmadm(1M).
+    qemu_extra_opts:
+        required: false
+        description:
+          - Additional qemu cmdline arguments for KVM guests.
+    quota:
+        required: false
+        description:
+          - Quota on zone filesystems (in MiBs).
+    ram:
+        required: false
+        description:
+          - Amount of virtual RAM for a KVM guest (in MiBs).
+    resolvers:
+        required: false
+        description:
+          - List of resolvers to be put into C(/etc/resolv.conf).
+    uuid:
+        required: false
+        description:
+          - UUID of the VM. Can either be a full UUID or C(*) for all VMs.
+    vcpus:
+        required: false
+        description:
+          - Number of virtual CPUs for a KVM guest.
+    vnc_port:
+        required: false
+        description:
+        - TCP port to listen of the VNC server. Or set C(0) for random,
+          or C(-1) to disable.
+    zfs_io_priority:
+        required: false
+        description:
+          - IO throttle priority value relative to other VMs.
+    zpool:
+        required: false
+        description:
+          - ZFS pool the VM's zone dataset will be created in.
+requirements:
+    - python >= 2.6
```

Comment:
There is very little precedent for that. I think it should be `python` as that is what most package managers call it too.